libfs/fs.go

```diff
@@ -277,6 +299,11 @@ func (fs *FS) OpenFile(filename string, flag int, perm os.FileMode) (
 		err = translateErr(err)
 	}()
 
+	err = fs.mkdirAll(path.Dir(filename), 0755)
```

Comment:
Have you considered the case where `path.Dir` would return `/` if there's only one element in a path with leading slash (e.g. `/blah`)? Looks like it's gonna cause problems in the `parts := strings.Split(leftover, "/")` later on?

Reply:
Good call thanks: #1175.